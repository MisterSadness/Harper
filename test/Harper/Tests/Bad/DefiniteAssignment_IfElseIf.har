f :: Integer -> Integer;
f n = {
  var (x :: Integer);
  var (result :: Integer) = 0;
  
  if (n == 42) {
    x := n;

    // x is definitey assigned.
    
    result += x;
  }
  else if (n < 42) {
    x := 0;

    // x is definitely assigned.

    result += x;
  }

  // x is not definitely assigned.

  result += x + n;
  return result;
};

main :: Integer;
main = f 42;