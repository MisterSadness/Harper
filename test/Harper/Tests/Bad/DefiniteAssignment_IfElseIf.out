Error: variable `x` is not definitely assigned at point of use.
During evaluation of:
  x
Located at line 23 column 13

Execution terminated with an error: type error.